LCLSGeneral/LCLSGeneral/POUs/Hardware/FB_LED_PWM.TcPOU

```diff
@@ -0,0 +1,56 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_LED_PWM" Id="{5fae4b24-b1d6-4ac5-a010-9aff4e2893db}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_LED_PWM
+(*
+    Reads an percentage of illumination determined by the user and converts it to a raw value
+    to send to the terminal output via the FB_AnalogOutput.
+    2022-5-03 Maarten Thomas-Bosum
+*)
+VAR
+	iIlluminatorINT AT %Q*: INT;
+
+    {attribute 'pytmc' := '
+        pv: PWR
+        field: ZNAM OFF
+        field: ONAM ON
+    '}
+    bLedPower AT %Q*: BOOL;
+
+    {attribute 'pytmc' := '
+        pv: CIL:PCT
+        EGU: %
+    '}
+    fIlluminatorPercent: LREAL;
+	
+    fbSetIllPercent: FB_AnalogOutput;
+END_VAR]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[// Illuminator conversion to percentage
+fbSetIllPercent(
+    fReal:=fIlluminatorPercent,
+    fSafeMax:=100,
+    fSafeMin:=0,
+    iTermBits:=14, // want to set scaling to 50% duty cycle = maximum raw bits sent to terminal output
+    fTermMax:=100,
+    fTermMin:=0,
+    iRaw=>iIlluminatorINT);
+
+IF  fIlluminatorPercent > 0 THEN
+	bLedPower := TRUE;
+ELSE
+	bLEDPower := FALSE;
+END_IF]]></ST>
+    </Implementation>
+    <Method Name="M_Set_IllPercentage" Id="{dbd69c00-c8e8-4492-8041-fbc83238a849}">
+      <Declaration><![CDATA[METHOD M_Set_IllPercentage : LREAL
+VAR_INPUT
+	IlluminatorPercent : LREAL;
+END_VAR
+]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[This^.fIlluminatorPercent := IlluminatorPercent;]]></ST>
+      </Implementation>
+    </Method>
+  </POU>
```

Comment:
Why have a method for this? Shouldn't `fIlluminatorPercent` be `VAR_INPUT`?

Reply:
That's a good idea, let me try that. This was my first iteration to "get things to work" so to say, so I saw that we had previously been setting variables via methods in the vacuum library.